### PR TITLE
feat: `environment.yaml` type

### PR DESCRIPTION
**Pull request by baszalmstra**
Adds a type for representing a `environment.yaml` based on https://github.com/conda/ceps/pull/81

### CODE REVIEW

File: test-data/environments/asymmetric_vqgan.environment.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - pytorch-lightning==1.4.2
+      - omegaconf==2.1.1
+      - test-tube>=0.7.5
+      - streamlit>=0.73.1
+      - einops==0.3.0
+      - torch-fidelity==0.3.0
+      - transformers==4.6.0
+      - torchmetrics==0.6
+      - academictorrents==2.3.3
+#    - -e git+https://github.com/CompVis/taming-transformers.git@master#egg=taming-transformers
```
Comment (ruben-arts) [inline test-data/environments/asymmetric_vqgan.environment.yaml:30]: Would be great if we could also read these when they are formatted like:
```
- -e "taming-transformers @ git+https://github.com/CompVis/taming-transformers.git@master#egg=taming-transformers"
```

**Review by ruben-arts — COMMENTED** [review]
Already looks great, one wish I have :D

File: test-data/environments/asymmetric_vqgan.environment.yaml (hunk 1 above)
Comment (baszalmstra) [reply test-data/environments/asymmetric_vqgan.environment.yaml:30]: We can :) We just parse it as a string!

Comment (baszalmstra) [reply test-data/environments/asymmetric_vqgan.environment.yaml]: I uncommented them to showcase this

**Pull request merged**